### PR TITLE
Change Basic Iterative Method (BIM)

### LINKED ISSUE #90: Change Basic Iterative Method (BIM)
The current implementation of the Basic Iterative Method asks for a value of `eps` but does not project to eps.

After discussing with @mathsinn we think that BIM should be implemented as a special case of PGD using no random initialisation (`num_random_init=0`) and `norm=np.inf`.

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

The current implementation of the Basic Iterative Method asks for a value of eps but does not project to eps.

After discussing with @mathsinn we think that BIM should be implemented as a special case of PGD using no random initialisation (num_random_init=0) and norm=np.inf.

Fixes #90 

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/iterative_method.py [hunk 1]
```diff
@@ -48,99 +46,14 @@ def __init__(self, classifier, norm=np.inf, eps=.3, eps_step=0.1, max_iter=100,
         :type eps: `float`
         :param eps_step: Attack step size (input variation) at each iteration.
         :type eps_step: `float`
-        :param max_iter: The maximum number of iterations.
```
Comment (mathsinn) [inline art/attacks/iterative_method.py]: I think this parameter should not be removed - can you please double check?

Comment (mathsinn) [reply art/attacks/iterative_method.py]: On the other hand, I think the parameter `norm` should be removed.

File: art/attacks/iterative_method.py [hunk 2]
```diff
@@ -48,99 +46,14 @@ def __init__(self, classifier, norm=np.inf, eps=.3, eps_step=0.1, max_iter=100,
...
-        :type max_iter: `int`
         :param targeted: Should the attack target one specific class
         :type targeted: `bool`
-        :param num_random_init: Number of random initialisations within the epsilon ball. For num_random_init=0
-            starting at the original input.
+        :param num_random_init: Number of random initialisations within the epsilon ball. For random_init=0 starting at
```
Comment (mathsinn) [inline art/attacks/iterative_method.py]: I think the `num_random_init` parameter should be removed.

File: art/attacks/iterative_method.py (hunk 1 above)
Comment (beat-buesser) [reply art/attacks/iterative_method.py]: I have resolved this with a38b34d8b5cfcbec46cee3b97fb6b9c98ebae283

File: art/attacks/iterative_method.py (hunk 2 above)
Comment (beat-buesser) [reply art/attacks/iterative_method.py]: I have resolved this with I have resolved this with a38b34d8b5cfcbec46cee3b97fb6b9c98ebae283

**Comment by beat-buesser** [comment]
@mathsinn Thank you for your review. During addressing your review requests I have noticed that the methods `set_params` also need to be updated and consistently applied in the `__init__` methods of `ProjectedGradientDescent` and `FastGradientMethod`

**Pull request merged**